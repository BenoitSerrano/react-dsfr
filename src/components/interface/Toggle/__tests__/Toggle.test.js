import renderer from 'react-test-renderer';
import { v4 as uuidv4 } from 'uuid';
import Toggle from '..';

jest.mock('uuid', () => ({
  v4: jest.fn(),
}));

describe('<Toggle />', () => {
  beforeEach(() => {
    uuidv4.mockImplementationOnce(() => 'xyx');
  });
  it('should render Toggle properly', () => {
    const onChange = jest.fn();
    const component = renderer
      .create(
        <Toggle
          checked
          onChange={onChange}
          label="Accepter les cookies"
          description="Activez-moi!"
          hasSeparator
          hasLabelLeft
          data-testid="toggle"
        />,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
