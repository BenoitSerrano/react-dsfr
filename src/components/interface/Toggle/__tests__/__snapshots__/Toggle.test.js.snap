// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toggle /> should render Toggle properly 1`] = `
<div
  className="fr-toggle fr-toggle--border-bottom fr-toggle--label-left"
  data-testid="toggle"
>
  <input
    className="fr-toggle__input"
    id="d8e868c0-7c3c-43ea-868d-5b7ac592c208"
    type="checkbox"
  />
  <label
    className="fr-toggle__label"
    data-fr-checked-label="Activé"
    data-fr-unchecked-label="Désactivé"
    htmlFor="d8e868c0-7c3c-43ea-868d-5b7ac592c208"
  >
    Accepter les cookies
  </label>
  <p
    className="fr-hint-text"
  >
    Activez-moi!
  </p>
</div>
`;
